doc/reset_factory/ResetFactoryHLD.md

```diff
@@ -0,0 +1,210 @@
+config-setup factory<br/>
+It will clear system logs and files.
+
+##### Keep-basic
+
+It will generate the default config_db.json and merge it with basic configurations from the current “config_db.json”.<br/>
+Command: config-setup factory keep-basic<br/>
+It will clear system logs and files.<br/>
+
+##### Keep-all-config
+It will keep the same configuration but will clear system logs and files.
+
+##### Only-config
+It will generate the default config_db.json file by running command:<br/>
+config-setup factory only-config<br/>
+It won't clear system logs and files.
+
+###  3.2 <a name='Config-Setup-Factory'></a>Config-setup Factory
+
+config-setup script provides the functionality of creating factory default configuration on demand.<br/>
+Currently it generates new config_db.json file using:<br/>
+```
+sonic-cfggen -H -k ${HW_KEY} --preset ${DEFAULT_PRESET}
+```
+
+Also, it allows the user to extend/replace the functionality by adding hooks to the config-setup directory.
+
+We will extend config-setup factory to support new factory type parameter (keep-basic).
+This parameter will be passed to the hook scripts so user can extend/replace those flows also.
+
+We will add a new conf file for config-setup: /etc/config-setup/config-setup.conf.
+The file will be imported in the script and it will store variables and configurations of the script.
+
+##### Keep-basic
+
+It will generate the default config_db.json merged with basic configurations from the current “config_db.json”<br/>
+The list of tables we will be stored in config-setup.conf:<br/>
+```
+KEEP_BASIC_TABLES=MGMT_PORT,MGMT_INTERFACE,PASSW_HARDENING
+```
+
+We will extend this list with the following tables after the merge of features:
+```
+SSH_SERVER,USER_TABLE,ROLE_TABLE
+```
+
+
+## 4 <a name='CLI'></a>CLI
+
+#### reset-factory
+```
+==============================================================================
+root@host:~$ sudo reset-factory --help
+ Usage:  reset-factory < keep-all-config | only-config | keep-basic >
+
+        Create factory default configuration and save it to
+        to /etc/sonic/config_db.json.
+        Clears logs, system files and reboot the system.
+
+        keep-all-config  - Preserves all configurations after boot. Logs and files will be deleted.
+        only-config      - Reset configurations to factory default. Logs and files will be preserved.
+        keep-basic       - Preserves basic configurations only after boot. Logs and files will be deleted.
+==============================================================================
+```
+
+#### config-setup factory
+```
+==============================================================================
+root@host:~$ sudo config-setup factory --help
+ Usage:  config-setup factory < keep-basic >
+
+         Create factory default configuration and save it to
+         to ${CONFIG_DB_JSON}.
+
+         keep-basic  - Preserves basic configurations only.
```

Comment:
keep-basic name doesnot look right, is it possible to have a default 'filter-list' with tables (e.g MGMT_PORT,MGMT_INTERFACE,PASSW_HARDENING), hope user can dynamically provide the tables to filter in a file as an option during factory-reset command execution..

Reply:
Please see my [comment](https://github.com/sonic-net/SONiC/pull/1231#discussion_r1088131084)
We need to take into account the user use case.
We want to allow the user to reset the system to factory settings without losing basic configurations.
I feel that "keep-basic" suits the use case and it explains the functionality of the command better than "filter-list".
Vendor can still extend/replace this list by changing "config-setup.conf" as explained in the design.